src/components/ThemeSwitch/ThemeSwitch.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { Option } from '@ui5/webcomponents-react/lib/Option';
+import { Select } from '@ui5/webcomponents-react/lib/Select';
+import { setTheme } from '@ui5/webcomponents-base/dist/config/Theme.js';
+import InformationDialog from '../InformationDialog/InformationDialog';
+
+const ThemeSwitch = ({ dialogRef, storedTheme = localStorage.getItem('reactSeedSelectedTheme') }) => {
+  const { t } = useTranslation();
+  setTheme(storedTheme ? storedTheme : 'sap_fiori_3');
+
+  const onChange = (event) => {
+    storedTheme = event.detail.selectedOption.dataset.value;
+    localStorage.setItem('reactSeedSelectedTheme', storedTheme);
+    setTheme(storedTheme);
+  };
+  const themeOptions = [
```

Comment:
Move this to outside the component. Example:
```js
const themeOptions = [...];
const style = { ... };

const ThemeSwitch = () => {
//...
}
```

On every prop or state change react re-renders the component, which means it executes the `ThemeSwitch` function again. That's why it's a good idea to make this function as lean as possible so you don't have to recreate the same constants over and over again.

Reply:
Thaaaanks bro, I'm making the updates!